### PR TITLE
PED1897-Add Rage encryption

### PR DESCRIPTION
**Pull request by Amrita42**
### PR creator: Description

The scope of this PR is to add Rage encryption software and tell users that this is the preferred method over GPG.   

### PR creator: Are there any relevant issues/feature requests?

[PED-1897](https://jira.suse.com/browse/PED-1897)

### PR creator: Which product versions do the changes apply to?

When opening a PR, check all versions of the documentation that your PR applies to.

- SLE 15/openSUSE Leap 15.x
  - [X] SLE 15 next/openSUSE Leap next *(current `main`, no backport necessary)*
  - [ ] SLE 15 SP4/openSUSE Leap 15.4
  - [ ] SLE 15 SP3/openSUSE Leap 15.3
  - [ ] SLE 15 SP2/openSUSE Leap 15.2
  - [ ] SLE 15 SP1
- SLE 12
  - [ ] SLE 12 SP5
  - [ ] SLE 12 SP4

### PR reviewer only: Have all backports been applied?

The doc team member merging your PR will take care of backporting to older documents.
When opening a PR, do *not* set the following check box.

- [ ] all necessary backports are done

### CODE REVIEW

**Comment by Amrita42** [comment]
[book-security_en.pdf](https://github.com/SUSE/doc-sle/files/11624278/book-security_en.pdf)

**Comment by Amrita42** [comment]
[book-security_en.pdf](https://github.com/SUSE/doc-sle/files/11624505/book-security_en.pdf)

File: xml/security_cryptofs.xml
Comment (Firstyear) [inline xml/security_cryptofs.xml:285]: This intro doesn't read very well? People will think small keys == insecure. Perhaps it should say "Rage has keys that are easy to exchange with other people, and has secure defaults to prevent accidental misuse or leaks of sensitive data".

File: xml/security_cryptofs.xml
Comment (Firstyear) [inline xml/security_cryptofs.xml:287]: The *recipient* must generate a key pair.

File: xml/security_cryptofs.xml
Comment (Firstyear) [inline xml/security_cryptofs.xml:291]: ssh rsa key or ssh ed25519 key specifically.

File: xml/security_cryptofs.xml
Comment (Firstyear) [inline xml/security_cryptofs.xml:296]: Should document this is the private key and should NOT be shared.

File: xml/security_cryptofs.xml
Comment (Firstyear) [inline xml/security_cryptofs.xml:304]: This is false, you name the encrypted output whatever the user wants with an extension they want.

File: xml/security_cryptofs.xml
Comment (Firstyear) [inline xml/security_cryptofs.xml:306]: The file can only be decrypted by the recipient who holds the corresponding private key.

File: xml/security_cryptofs.xml
Comment (Firstyear) [inline xml/security_cryptofs.xml:318]: ssh *keys*.

File: xml/security_cryptofs.xml
Comment (Firstyear) [reply xml/security_cryptofs.xml:291]: given you have an ssh section later, you likely don't need this comment about ssh keys here.

File: xml/security_cryptofs.xml
Comment (Firstyear) [inline xml/security_cryptofs.xml:323]: This statement doesn't make sense and isn't factually correct IMO.

**Review by Firstyear — CHANGES_REQUESTED** [review]

File: xml/security_cryptofs.xml
Comment (Amrita42) [reply xml/security_cryptofs.xml:291]: Yeah , good point , removed this line.

**Comment by Amrita42** [comment]
@Firstyear Implemented review feedback, if all looks good , please add **lgtm** label

File: xml/security_cryptofs.xml
Comment (cwickert) [inline xml/security_cryptofs.xml:360]: This is kind of redundant. Simplfy to:

```suggestion
  <para>
   You can use the SSH (Secure Shell Protocol) keys in place of the Rage keys. Rage
   supports <emphasis>ssh-rsa</emphasis> and
   <emphasis>ssh-ed25519</emphasis> public keys, and decrypting with the
   respective private key file. <command>ssh-agent</command> is not supported.
  </para>
```

File: xml/security_cryptofs.xml
Comment (cwickert) [inline xml/security_cryptofs.xml:388]: See above, either `<sect3>` or restructure.

**Review by cwickert — COMMENTED** [review]
I took the liberty to reformat and fix some obvious style issues in 7abfafacca3912da477282078fda7e919076eae1. Please have a look at the commit, it's too much to list everything here.

The rest is subject to discussion. I very much like the content, but we should start with a brief introduction to explain

* what are and rage are (while are is in Go, rage is the rust implementation)
* and how to install it (zypper install rage-encryption).

**Review by cwickert — CHANGES_REQUESTED** [review]
I took the liberty to reformat and fix some obvious style issues in 7abfafacca3912da477282078fda7e919076eae1. Please have a look at the commit, it's too much to list everything here.

The rest is subject to discussion. I very much like the content, but we should start with a brief introduction to explain

* what are and rage are (while are is in Go, rage is the rust implementation)
* and how to install it (zypper install rage-encryption).

File: xml/security_cryptofs.xml
Comment (Firstyear) [reply xml/security_cryptofs.xml:290]: You need to say sender/recipient because else the *sender* may think they need to generate the keys. Previously the wording was confusing and by not distinguishing made it very confusing.

File: xml/security_cryptofs.xml
Comment (Firstyear) [reply xml/security_cryptofs.xml:312]: I don't think that's needed, because then there is a whole discussion of "do you keep the private key long term or not" and "once you decrypt a file how to do you make sure it's content never leaks" etc. Rage protects the file from point A to B, not to protect a file locally for all of time. There also a whole discussion there about key longevity too. As well, once you decrypt the file then you have the decrypted file ...

File: xml/security_cryptofs.xml
Comment (Firstyear) [reply xml/security_cryptofs.xml:287]: It's already got a PM ticket, and already had review and approval from security.

**Review by Firstyear — CHANGES_REQUESTED** [review]

**Comment by Amrita42** [comment]
> I took the liberty to reformat and fix some obvious style issues in [7abfafa](https://github.com/SUSE/doc-sle/commit/7abfafacca3912da477282078fda7e919076eae1). Please have a look at the commit, it's too much to list everything here.
> 
> The rest is subject to discussion. I very much like the content, but we should start with a brief introduction to explain
> 
> * what are and rage are (while are is in Go, rage is the rust implementation)
> * and how to install it (zypper install rage-encryption).

@cwickert thanks! but please hold off any reviews till I request for one. I was going to go through post the tech review for a style check review as well.

File: xml/security_cryptofs.xml
Comment (Amrita42) [reply xml/security_cryptofs.xml]: I agree with retaining recipient( Since I ve taken the maintainer role for the Security/Hardening Book , I will go going through in-depth and looking at the existing content incl GPG).

**Comment by Amrita42** [comment]
@Firstyear Could you please take a final ook from the tech perspective if all looks good. 
I am not implementing the style changes as that is the next step.  Also added a sentence to use zypper to install rage encryption.

**Review by Firstyear — CHANGES_REQUESTED** [review]

File: xml/security_cryptofs.xml
Comment (Amrita42) [reply xml/security_cryptofs.xml:336]: re-phrased

**Comment by Amrita42** [comment]
[Uploading book-security_en.pdf…]()

@Firstyear Feedback implemented.

**Review by Firstyear — APPROVED** [review]

**Comment by Amrita42** [comment]
lgtm

File: xml/security_cryptofs.xml
Comment (Amrita42) [reply xml/security_cryptofs.xml:307]: implemented

File: xml/security_cryptofs.xml
Comment (Amrita42) [reply xml/security_cryptofs.xml]: Retained as is.

File: xml/security_cryptofs.xml
Comment (Amrita42) [reply xml/security_cryptofs.xml:295]: Used variablelist element. Thanks.

File: xml/security_cryptofs.xml
Comment (Amrita42) [reply xml/security_cryptofs.xml]: Used variablelist element

**Pull request merged**
